File: app/src/main/java/com/lzm/lightLive/http/request/hy/HyBasicConverter.java
package com.lzm.lightLive.http.request.hy;

import android.util.Log;
import com.google.gson.Gson;
import com.google.gson.JsonObject;
import com.google.gson.reflect.TypeToken;
import com.lzm.lightLive.http.CustomGsonConvertFactory;
import com.lzm.lightLive.http.bean.hy.HyRoom;
import com.lzm.lightLive.util.CommonUtil;
import java.io.IOException;
import okhttp3.ResponseBody;

public class HyBasicConverter extends CustomGsonConvertFactory.ResponseConverter {

    @Override
    protected String convertResponse(ResponseBody value) throws IOException {
        String response = value.string();
        JsonObject obj = new Gson().fromJson(response, JsonObject.class);
        Log.e("convertResponse: ", obj.toString() );


        if(obj.has("data") && obj.get("data") != null) {
            HyRoom roomInfo = new Gson().fromJson(obj.get("data"), HyRoom.class);
            String roomName = roomInfo.getHyLiveData().getRoomName();
            long userCount = roomInfo.getHyLiveData().getUserCount();
            long profileRoom = roomInfo.getHyLiveData().getProfileRoom();
            String screenshot = roomInfo.getHyLiveData().getScreenshot();
            String fullName = roomInfo.getHyLiveData().getGameFullName();

            if(roomInfo.getHyStream() != null
                    && roomInfo.getHyStream().getFlv() != null
                    && roomInfo.getHyStream().getFlv().getMultiLine().get(0) != null) {
                String url = roomInfo.getHyStream().getFlv().getMultiLine().get(0).getUrl();
                roomInfo.setLiveStreamUri(url);
            }

            String nick = roomInfo.getHyProfileInfo().getNick();
            String avatar = roomInfo.getHyProfileInfo().getAvatar();
            int activityCount = roomInfo.getHyProfileInfo().getActivityCount();

            roomInfo.setHostName(nick);
            roomInfo.setRoomName(roomName);
            roomInfo.setCateName(fullName);
            roomInfo.setHostAvatar(avatar);
            roomInfo.setHeatNum(userCount);
            roomInfo.setThumbUrl(screenshot);
            roomInfo.setFansNum(activityCount);
            roomInfo.setStreamStatus(CommonUtil.generateLiveStatus(roomInfo.getHyRealLiveStatus()));
            roomInfo.setRoomId(String.valueOf(profileRoom));
            obj.remove("data");
            obj.add("data", new Gson().toJsonTree(roomInfo, new TypeToken<HyRoom>() {}.getType() ));
            response = obj.toString();
        }
        return response;
    }
}

File: app/src/main/java/com/lzm/lightLive/http/request/hy/HyBasicConverter.kt
package com.lzm.lightLive.http.request.hy

import android.util.Log
import com.google.gson.Gson
import com.google.gson.JsonObject
import com.google.gson.reflect.TypeToken
import com.lzm.lightLive.http.CustomGsonConvertFactory.ResponseConverter
import com.lzm.lightLive.http.bean.hy.HyRoom
import com.lzm.lightLive.util.CommonUtil
import okhttp3.ResponseBody
import java.io.IOException

class HyBasicConverter : ResponseConverter() {

    @Throws(IOException::class)
    override fun convertResponse(value: ResponseBody): String {
        var response = value.string()
        val obj = Gson().fromJson(response, JsonObject::class.java)
        Log.e("convertResponse: ", obj.toString())
        if (obj.has("data") && obj["data"] != null) {
            val roomInfo = Gson().fromJson(obj["data"], HyRoom::class.java)
            val roomName = roomInfo.hyLiveData?.roomName
            val userCount = roomInfo.hyLiveData!!.userCount
            val profileRoom = roomInfo.hyLiveData?.profileRoom
            val screenshot = roomInfo.hyLiveData?.screenshot
            val fullName = roomInfo.hyLiveData?.gameFullName
            if (roomInfo.hyStream != null
                && roomInfo!!.hyStream?.flv != null
                && roomInfo.hyStream?.flv?.multiLine?.get(0) != null) {
                val url = roomInfo.hyStream!!.flv?.multiLine?.get(0)?.url
                roomInfo.liveStreamUriHigh = url
            }
            val nick = roomInfo.hyProfileInfo?.nick
            val avatar = roomInfo.hyProfileInfo?.avatar
            val activityCount = roomInfo.hyProfileInfo?.activityCount
            roomInfo.hostName = nick
            roomInfo.roomName = roomName
            roomInfo.cateName = fullName
            roomInfo.hostAvatar = avatar
            roomInfo.heatNum = userCount
            roomInfo.thumbUrl = screenshot
            roomInfo.fansNum = activityCount!!.toLong()
            roomInfo.streamStatus = CommonUtil.generateLiveStatus(roomInfo.hyRealLiveStatus)
            roomInfo.roomId = profileRoom.toString()
            obj.remove("data")
            obj.add("data", Gson().toJsonTree(roomInfo, object : TypeToken<HyRoom?>() {}.type))
            response = obj.toString()
        }
        return response
    }
}